### PR TITLE
Add Emilia Training Recipe for Llasa (cosyvoice2 token)

### PR DESCRIPTION
**Pull request by yuekaizhang**
Inspired by [Llasa](https://arxiv.org/abs/2502.04128), this PR enables continued pretraining of the Qwen2 LLM for the CosyVoice2 semantic token prediction task. 

The predicted semantic tokens can be used to generate audio with either the CosyVoice2 pretrained U-Net model or the DIT model [PR](https://github.com/k2-fsa/icefall/pull/1880).

|        LLM Model                               | Flow matching Model | Seed-TTS test_zh CER  | Comment                                           |
|---------------------------------------|----------|-----------|--------|
| pretrained cosyvoice2 0.5B | f5-tts-small (wenetspeech4tts 7k hours) |  1.79% (16 steps)   | See [PR](https://github.com/k2-fsa/icefall/pull/1880)|
| llasa_cosyvoice2_token 0.5B (Emilia_ZH 50k hours) | f5-tts-small (wenetspeech4tts 7k hours) |  1.89% (16 steps)   | |

### CODE REVIEW

**Comment by danpovey** [comment]
This seems like good work, and it's nice that you want to include it in our collection of recipes.
I also find it quite interesting.  But I'm trying to come up with a good justification why it should be included here,
other than the fact that we are also interested in the TTS task right now.  I.e. are we OK with icefall being a collection of
recipes even in cases where they have very little in common?
(BTW I notice that the instructions direct the user installs the k2 package, but I doubt this is actually needed).
Regardless of whether we merge it (and I'm open to input from our team members and others on this issue), I'm happy to have the pull request left here as an accessible place for discussion about this recipe and so that we can easily find it.

**Comment by yuekaizhang** [comment]
> This seems like good work, and it's nice that you want to include it in our collection of recipes. I also find it quite interesting. But I'm trying to come up with a good justification why it should be included here, other than the fact that we are also interested in the TTS task right now. I.e. are we OK with icefall being a collection of recipes even in cases where they have very little in common? (BTW I notice that the instructions direct the user installs the k2 package, but I doubt this is actually needed). Regardless of whether we merge it (and I'm open to input from our team members and others on this issue), I'm happy to have the pull request left here as an accessible place for discussion about this recipe and so that we can easily find it.

Thank you for your feedback! Indeed, the structure of this PR differs from other recipes in Icefall. Initially, I planned to implement it using Lhotse and Icefall training loops. However, I found it simpler to use the Hugging Face dataset and trainer since it's a language model token prediction task. 

I have added a [Not for merge] tag so that people can still reference the results in the PR.

File: egs/emilia/TTS/README.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+# Introduction
+
+[**Emilia**](https://huggingface.co/datasets/amphion/Emilia-Dataset) starts with over 101k
+hours of speech across six languages, covering a wide range of speaking styles to enable more natural and spontaneous speech generation.
+
+See https://arxiv.org/pdf/2407.05361.
+
+> [!CAUTION]
+> The next-gen Kaldi framework provides tools and models for generating high-quality, synthetic speech (Text-to-Speech, TTS).
```
[the remark this answers is not in the document]
Comment (yuekaizhang) [reply egs/emilia/TTS/README.md]: Yeah, I copied from libritts recipe here https://github.com/k2-fsa/icefall/tree/master/egs/libritts/TTS#readme. Deleted now.
